core/trino-main/src/main/java/io/trino/sql/planner/MergePartitioningHandle.java

```diff
@@ -94,6 +96,28 @@ public NodePartitionMap getNodePartitioningMap(Function<PartitioningHandle, Node
         return optionalInsertMap.orElseGet(optionalUpdateMap::orElseThrow);
     }
 
+    public FaultTolerantPartitioningScheme getFaultTolerantScheme(Function<PartitioningHandle, FaultTolerantPartitioningScheme> getMap)
```

Comment:
nit: `getMap` -> `createPartitiningScheme`

Reply:
Good idea to change the name, but I decided `partitionMapFactory` was a bit better.